### ISSUE TITLE
Allow specifying enableServiceLinks configuration in PodTemplate

### ISSUE THREAD

**Issue by artifactron** [strimzi/strimzi-kafka-operator#4800]
**Is your feature request related to a problem? Please describe.**
We're running a Kubernetes cluster with 1k+ `Service` objects, resulting in 5k+ environment variables being injected to pods if `enableServiceLinks` configuration is set to `true`, which is also its default value.

Zookeeper and Kafka pods seem can't handle that tremendous amount of environment variables as they keep crashing with no meaningful explanation in the log. Changing `enableServiceLinks` to `false` in the Kafka `StatefulSet` object (by temporarily pausing the operator) fixes the issue, but sure, if the operator is resumed, the configuration will be reverted and Kafka will start to crash.

**Describe the solution you'd like**
Allow configuring `enableServiceLinks` in `PodTemplate` so we can set it to `false`, resulting in fewer environment variables being injected to the Strimzi-managed pods, which results in working Kafka clusters on the specified cluster.

**Describe alternatives you've considered**
Allow more flexible schema of `PodTemplate` so more aspects of Strimzi-managed pods can be configured. [An example of an operator that leverages this flexible schema is **elastic-operator**](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-customize-pods.html).

**Additional context**
_N/A_

**Comment by scholzj** [comment]
I do not think we plan to support something like the Elastic operator has. But I will have a look at the `enableServiceLinks`.
